### PR TITLE
skylog: Get tree sharding info from BuildJob

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
The worker Cloud Function now takes sharding info from the passed-in event.

### CODE REVIEW

File: skylog/storage/gcp/cloudfunc/go.mod [hunk 1]
```diff
@@ -11,6 +11,6 @@ require (
 	github.com/go-toolsmith/pkgload v1.0.0 // indirect
 	github.com/go-toolsmith/typep v1.0.0 // indirect
 	github.com/golang/protobuf v1.3.2-0.20190517061210-b285ee9cfc6c
-	github.com/google/trillian v1.2.2-0.20190619133759-4511faac1a5c
```
Comment (AlCutter) [inline skylog/storage/gcp/cloudfunc/go.mod]: Are these module changes expected too?

File: skylog/storage/gcp/cloudfunc/worker.go [hunk 2]
```diff
@@ -71,14 +71,27 @@ func BuildSubtree(ctx context.Context, msg BuildMessage) error {
...
 	_, err = bw.Process(ctx, cJob)
 	return err
 }
 
+func treeOpts(job *pb.BuildJob) (cs.TreeOpts, error) {
+	ts := job.GetTreeSharding()
+	if ts == nil {
+		return cs.TreeOpts{}, errors.New("missing tree sharding info")
+	} else if ts.Levels <= 0 || ts.Shards <= 0 {
+		return cs.TreeOpts{}, errors.New("invalid tree sharding info")
```
Comment (AlCutter) [inline skylog/storage/gcp/cloudfunc/worker.go]: `fmt.Errorf()` with the invalid config? (to help the person debugging)

File: skylog/storage/gcp/cloudfunc/go.mod (hunk 1 above)
Comment (pavelkalinnikov) [reply skylog/storage/gcp/cloudfunc/go.mod]: Yes, it picks up new additions to the `BuildJob` proto.

File: skylog/storage/gcp/cloudfunc/worker.go (hunk 2 above)
Comment (pavelkalinnikov) [reply skylog/storage/gcp/cloudfunc/worker.go]: Done.

**Pull request merged**